### ISSUE TITLE
@XmlAttribute is ignored if @XmlAccessorType(XmlAccessType.NONE) is used by swagger-core 1.5.16 and master

### ISSUE THREAD

**Issue by dranuhl** [swagger-api/swagger-core#2487]
If the `@XmlAccessorType(XmlAccessType.NONE)` is used on some model class and explicit annotation of fields or methods with the `@XmlAttribute` annotation has no effect.

The issue occurs using swagger-core 1.5.16 and even the most recent one within master seems to have the same code base in this respect.
Problem is likely the `io.swagger.jackson.ModelResolver.ignore(Annotated,XmlAccessorType,String,Set)` method, which just honors the `@XmlElement` annotation in this case, but not the `@XmlAttribute` one.

Example code can be found at https://github.com/dranuhl/swagger-bug-examples using swagger 1.5.16 for Jersey 1.x and 2.x.

See also [Swagger Online forum](https://swagger.io/forum/#!msg/swagger-swaggersocket/vGOJ3Hby0E4/PVov28dmAQAJ) report.

**Comment by dranuhl** [comment]
The following diff resolves the above described issue with v1.5.16:
```
diff --git a/modules/swagger-core/src/main/java/io/swagger/jackson/ModelResolver.java b/modules/swagger-core/src/main/java/io/swagger/jackson/ModelResolver.java
index 035d49f..ac36489 100644
--- a/modules/swagger-core/src/main/java/io/swagger/jackson/ModelResolver.java
+++ b/modules/swagger-core/src/main/java/io/swagger/jackson/ModelResolver.java
@@ -54,6 +54,7 @@ import javax.validation.constraints.Pattern;
 import javax.validation.constraints.Size;
 import javax.xml.bind.annotation.XmlAccessType;
 import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
 import javax.xml.bind.annotation.XmlElement;
 import javax.xml.bind.annotation.XmlRootElement;
 import javax.xml.bind.annotation.XmlSchema;
@@ -617,7 +618,8 @@ public class ModelResolver extends AbstractModelConverter implements ModelConver
             return false;
         }
         if (xmlAccessorTypeAnnotation.value().equals(XmlAccessType.NONE)) {
-            if (!member.hasAnnotation(XmlElement.class)) {
+            if (!member.hasAnnotation(XmlElement.class) &&
+                !member.hasAnnotation(XmlAttribute.class)) {
                 return true;
             }
         }

```

**Comment by dranuhl** [comment]
See also pull request #2409, which even includes some tests.

**Comment by dranuhl** [comment]
This issue duplicates #1847.

**Comment by frantuma** [comment]
merged #2409
